docs/hibernating-clusters.md

```diff
@@ -0,0 +1,121 @@
+# Hibernating Clusters
+
+## Overview
+
+Starting with OpenShift version 4.4.8, clusters can be stopped and started by simply shutting
+down their machines and starting them back up. The only additional requirement to bring a cluster
+back up from a stopped state is that CSRs be approved for the cluster's nodes when certificates
+have expired since the cluster was active.
+
+Hive can automate the process of stopping/starting clusters via its API by allowing the user to
+set a the desired state of the cluster in the ClusterDeployment spec. Both API and controller
+changes are required to support this feature.
+
+## API Changes
+
+The ClusterDeploymentSpec should allow setting whether machines are in a running state or in
+a hibernating state.
+
+```go
+type ClusterDeploymentSpec struct {
+  // ... other fields
+
+  // Hibernating indicates whether a cluster should be hibernating. Default is false.
+  Hibernating  `json:"hibernating,omitempty"`
```

Comment:
I assume that this is missing a `bool`. I think we should make this an enum string so that we have the ability to support more modes in the future.

Reply:
yes, it is a missing bool. I thought about the enum, but struggled to think about other modes that would make sense. Not opposed to changing it though.

Reply:
I think the general guidance is to avoid bool because it locks you in to binary state. However, I cannot think of any other modes either.